### DISCUSSION TITLE
Chocolatey failed to download app from my source(via powershell)

### DISCUSSION THREAD

**Discussion by Timmy5818** [chocolatey/choco#2711] [General] [answered]
I put a software on my Chocolatey Package Repository and want to install it, but the downloaded URL is from my Laravel website, the website needs to be logged in to download files, so Chocolatey will have an error and cannot be installed, how can I install this application?

I search some solution following, but still can't work it:

[How to Download a File with PowerShell](https://lazyadmin.nl/powershell/download-file-powershell/?unapproved=6092&moderation-hash=0e2161927d420a72ae746d97ba809d2a#comment-6092)

The above website mentioned that if the website needs to log in to download, you can use the sample code in the website, but I still can't download it.

Install the chocolatey software error message with CMD:
![image](https://user-images.githubusercontent.com/87130761/166399994-57205159-7a2a-4747-9ecd-ac5dcdd41e2e.png)

**Comment by TheCakeIsNaOH** [comment]
Why not embed the software into the package, instead of downloading it during install?

**Comment by Timmy5818** [comment]
> Why not embed the software into the package, instead of downloading it during install?

How can I embed the sofware into the package , is there any reference?

**Comment by TheCakeIsNaOH** [comment, marked as answer]
There are instructions in the default package template (what you get by default when you run `choco new <name>`).

To be more specific than that, it depends on what type of app you are packaging. For example, if you need to extract an archive, use `Get-ChocolateyUnzip` to extract an embedded file.

**Comment by Timmy5818** [comment]
Thank you, I'll go to check out. 

If there 's any question, I might ask you again.

**Comment by Timmy5818** [comment]
In addition, I would like to ask why the installation of the 7-Zip program I placed in Chocolatey Package Repository shows that the installation failed, but it has actually been installed successfully. What parameters need to be set?

Below is the error screen:
![image](https://user-images.githubusercontent.com/87130761/166615573-170e71cc-ff94-46d7-8667-7db3530300cc.png)

**Reply by TheCakeIsNaOH** [reply]
In this screenshot, the command being run is actually attempting to install two packages, `nas_server_test` and `21.07`

The `nas_server_test` package installed, and the  `21.07` package was not found.

If you are trying to specify the version of the `nas_server_test`, you need to use `--version=21.07`

**Reply by Timmy5818** [reply]
> In this screenshot, the command being run is actually attempting to install two packages, `nas_server_test` and `21.07`
> 
> The `nas_server_test` package installed, and the `21.07` package was not found.
> 
> If you are trying to specify the version of the `nas_server_test`, you need to use `--version=21.07`

Thank you, I found that this problem is installed to two packages, here I only need to install nas_server_test.
